pep-0458.txt

```diff
@@ -819,7 +819,13 @@ include a version number in their filename:
   VERSION_NUMBER.ROLENAME.json,
     where VERSION_NUMBER is an incrementing integer, and ROLENAME is one of the
     top-level metadata roles -- *root*, *snapshot* or *targets* -- or one of
-    the delegated targets roles -- *bins* or *bin-n*.
+    the delegated targets roles -- *bins* or *bin-n*. The version number for
+    *snapshot*, *targets*, *bins*, and *bin-n* roles may eventually loop back to 0 as long
+    as the previous 0-version metadata has expired before this occurs. To
+    prevent confusion, we suggest retiring the oldest metadata for each non-root
+    metadata type when 0.5*MAX_INT versions are in use. Root metadata can
```

Comment:
Will the reader know what "0-version metadata" means? To me the term is confusing. And do you mean "expired" or "retired"? In the ["Cleaning up old metadata" section](https://github.com/secure-systems-lab/peps/blob/1st-draft/pep-0458.txt#L924) we explain how metadata may be retired if it has become obsolete, which very likely happens before it has expired.

On the other hand, we must not retire metadata which has not yet become obsolete, even if its version number exceeds the `0.5*MAX_INT` threshold. But that's very unlikely to happen. If we clean up metadata once per hour, it would need `0.5*MAX_INT` metadata bumps within an hour to override.

But the stay on the safe side, maybe we can leave the details to the implementer and just say something along the lines...

```
To avoid integer overflows, the version numbers for *snapshot*, *targets*,
*bins*, and *bin-n* roles may be reset at regular intervals, under the
condition that no version number is used by two or more metadata files for a
role at the same time (cf. `Cleaning up old metadata`_ ). We suggest to retire
old metadata before `0.5*MAX_INT` versions are in use.
```